modules/OAuth2AuthCodes/views/view.authorize.php

```diff
@@ -0,0 +1,148 @@
+* this program; if not, see http://www.gnu.org/licenses or write to the Free
+* Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+* 02110-1301 USA.
+*
+* You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+* SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+*
+* The interactive user interfaces in modified source and object code versions
+* of this program must display Appropriate Legal Notices, as required under
+* Section 5 of the GNU Affero General Public License version 3.
+*
+* In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+* these Appropriate Legal Notices must retain the display of the "Powered by
+* SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+* reasonably feasible for technical reasons, the Appropriate Legal Notices must
+* display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+*/
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+use League\OAuth2\Server\AuthorizationServer;
+use Api\V8\OAuth2\Entity\UserEntity;
+
+/**
+* Class Oauth2AuthCodesViewAuthorize
+*/
+class Oauth2AuthCodesViewAuthorize extends SugarView
+{
+    /**
+    * @var array $options
+    * Options for what UI elements to hide/show/
+    */
+    public $options = array(
+        'show_header' => false,
+        'show_title' => false,
+        'show_subpanels' => false,
+        'show_search' => false,
+        'show_footer' => false,
+        'show_javascript' => true,
+        'view_print' => false,
+    );
+
+    public function display()
+    {
+        $app = new \Slim\App(\Api\Core\Loader\ContainerLoader::configure());
+        $server   = $app->getContainer()->get(AuthorizationServer::class);
+        $request  = $app->getContainer()->get('request');
+        $response = $app->getContainer()->get('response');
+
+        if(strpos($_SERVER['HTTP_REFERER'], 'action=Login') !== false){
+            $_SESSION['oauth2authcode_logout'] = TRUE;
+        } else {
+            unset($_SESSION['oauth2authcode_logout']);
+        }
+
+        if(!isset($_SESSION['oauth2authcode']) ||
+        $request->getParam('confirmed') == null ||
+        $request->getParam('session_id') == null ||
+        $request->getParam('session_id') != session_id() ||
+        $request->getParam('oauth2authcode_hash') == null ||
+        $request->getParam('oauth2authcode_hash') != $_SESSION['oauth2authcode_hash'])
+        {
+            try {
+                $authRequest = $server->validateAuthorizationRequest($request);
+            } catch (League\OAuth2\Server\Exception\OAuthServerException $exception) {
+                sugar_die($GLOBALS['mod_strings']['LBL_INVALID_REQUEST'].": ".$exception->getMessage());
+            }
+
+            global $current_user;
+            $authRequest->setUser(new UserEntity($current_user->id)); // an instance of UserEntityInterface
```

Comment:
```$authRequest``` might still be undefined at this point.

Reply:
Added 
```
if ($authRequest === null) {
    throw new \InvalidArgumentException($GLOBALS['mod_strings']['LBL_INVALID_REQUEST']);
}
```
to make sure, ```$authRequest``` is defined. Is that sufficient?